sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -146,21 +154,25 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
      * @param activationOperation The activation operation to be called by the {@link Poller} instance before
      *     calling {@code pollOperation}. It can be {@code null} which will indicate to the {@link Poller} that
      *     {@code pollOperation} can be called straight away.
+     * @param fetchResultOperation The operation to be called to fetch final result after polling has been completed.
      * @param cancelOperation Cancel operation if cancellation is supported by the service. If it is {@code null}, then
      *     the cancel operation is not supported.
      * @throws IllegalArgumentException if {@code pollInterval} is less than or equal to zero and if
      *      {@code pollInterval} or {@code pollOperation} are {@code null}
      */
     public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation,
-                  Supplier<Mono<T>> activationOperation, Consumer<Poller<T>> cancelOperation) {
+                  Supplier<Mono<R>> fetchResultOperation, Supplier<Mono<T>> activationOperation,
+                  Function<Poller<T, R>, Mono<T>> cancelOperation) {
 
         Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
+        Objects.requireNonNull(fetchResultOperation, "'fetchResultOperation' cannot be null.");
```

Comment:
What do you expect it to be in cases where we don't care about a fetch operation?

Reply:
() -> Mono.empty
This should be enough for fetch operation, if R is Void in Poller<T,R>